addons/common/functions/fnc_setLampState.sqf

```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Sets the state (on/off) of the given lamp object.
+ *
+ * Arguments:
+ * 0: Lamp <OBJECT>
+ * 1: State <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [_lamp, false] call zen_common_fnc_setLampState
```

Comment:
I think there is already a BI function for this. Any reason for adding our own?

Reply:
Do a comparison of them. General rule of thumb for ACEifying anything is to not rely on BI functions, they may be written in a worse way, without the given framework for debugging, or have additional checks that slow it down and are not necessary.

Reply:
BI's
```sqf
params [
	["_lamp",objnull,[objnull]],
	["_state",true,[true]]
];
_lamp switchlight (["off","on"] select _state);
```

It uses `switchLight` instead of `setHit` but it seems that in the latest A3 updates `switchLight` has been extended to allow it to be used on street lamps too. Don't see the point of using ZEN's implementation.

Reply:
Own function is better here, it will handle destroyed lights.

Reply:
I'm not trying to advocate for BI's code, but handling destroyed lights might not be what the user intends. A player might shoot out lights and then using this module would repair them all back to normal. A choice would be good.